From 3b1d63ff8972e782f537edae8e8f2e00294f8065 Mon Sep 17 00:00:00 2001
From: Meng Mingming <mingming.meng@nxp.com>
Date: Wed, 13 Dec 2017 18:01:41 +0800
Subject: [PATCH 3088/5242] MLK-17204 drm/imx: dpu: Add suspend/resume support
 for dpu-blit

commit  047992b440db11cf0e35e936d2c5789d9be0da3a from
https://source.codeaurora.org/external/imx/linux-imx.git

Add suspend/resume support for dpu bliteng device.

Signed-off-by: Meng Mingming <mingming.meng@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dpu/dpu-blit.c |   34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/drivers/gpu/drm/imx/dpu/dpu-blit.c b/drivers/gpu/drm/imx/dpu/dpu-blit.c
index fae1d133..164f651 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-blit.c
+++ b/drivers/gpu/drm/imx/dpu/dpu-blit.c
@@ -256,9 +256,43 @@ static int dpu_bliteng_remove(struct platform_device *pdev)
 	return 0;
 }
 
+#ifdef CONFIG_PM_SLEEP
+static int dpu_bliteng_suspend(struct device *dev)
+{
+	struct dpu_bliteng *dpu_bliteng = dev_get_drvdata(dev);
+	int ret;
+
+retry:
+	ret = dpu_be_get(dpu_bliteng);
+	if (ret == -EBUSY)
+		goto retry;
+
+	dpu_be_wait(dpu_bliteng);
+
+	dpu_be_put(dpu_bliteng);
+
+	dpu_bliteng_fini(dpu_bliteng);
+
+	return 0;
+}
+
+static int dpu_bliteng_resume(struct device *dev)
+{
+	struct dpu_bliteng *dpu_bliteng = dev_get_drvdata(dev);
+
+	dpu_bliteng_init(dpu_bliteng);
+
+	return 0;
+}
+#endif
+
+static SIMPLE_DEV_PM_OPS(dpu_bliteng_pm_ops,
+			 dpu_bliteng_suspend, dpu_bliteng_resume);
+
 struct platform_driver dpu_bliteng_driver = {
 	.driver = {
 		.name = "imx-drm-dpu-bliteng",
+		.pm = &dpu_bliteng_pm_ops,
 	},
 	.probe = dpu_bliteng_probe,
 	.remove = dpu_bliteng_remove,
-- 
1.7.9.5

